confgenerator/self_logs.go

```diff
@@ -123,55 +121,47 @@ func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Compon
 			},
 		},
 	}.Components(ctx, fluentBitSelfLogsTag, "fluent-bit-self-log-regex-parsing")...)
+	// Disables sending fluent-bit debug logs to Cloud Logging due to endless spam.
+	// TODO: Remove when b/272779619 is fixed.
+	if logLevel == "debug" {
+		out = append(out, []fluentbit.Component{
+			{
+				Kind: "FILTER",
+				Config: map[string]string{
+					"Name":    "grep",
+					"Match":   fluentBitSelfLogsTag,
+					"Exclude": "severity debug",
+				},
+			},
+		}...)
+	}
 	return out
 }
 
-type selfLogTranslationEntry struct {
-	regexMatch string
-	message    string
-	code       string
-}
-
-var selfLogTranslationList = []selfLogTranslationEntry{
-	{
-		regexMatch: `\[error\]\s\[lib\]\sbackend\sfailed`,
-		message:    fmt.Sprintf("Ops Agent logging pipeline failed, Code: LogPipelineErr, Documentation: %s", troubleshootFindInfoURL),
-		code:       "LogPipelineErr",
-	},
-	{
-		regexMatch: `\[error\]\s\[parser\]\scannot\sparse`,
-		message:    fmt.Sprintf("Ops Agent failed to parse logs, Code: LogParseErr, Documentation: %s", troubleshootFindInfoURL),
-		code:       "LogParseErr",
-	},
-}
-
-func generateSelfLogsSamplingComponents(ctx context.Context) []fluentbit.Component {
+func generateFilterSelfLogsSamplingComponents(ctx context.Context) []fluentbit.Component {
 	out := make([]fluentbit.Component, 0)
 
-	for _, m := range selfLogTranslationList {
+	for _, m := range healthchecks.FluentBitSelfLogTranslationList {
 		// This filter samples specific fluent-bit logs by matching with regex and re-emits
 		// an `ops-agent-health` log.
 		out = append(out, fluentbit.Component{
 			Kind: "FILTER",
 			Config: map[string]string{
 				"Name":  "rewrite_tag",
 				"Match": fluentBitSelfLogsTag,
-				"Rule":  fmt.Sprintf(`message %s %s true`, m.regexMatch, healthLogsTag),
+				"Rule":  fmt.Sprintf(`message %s %s true`, m.RegexMatch, healthLogsTag),
 			},
 		})
 		// This filter sets the appropiate health code to the previously sampled logs. The `code` is also
 		// set to the `message` field for later translation in the pipeline.
-		// The current fluent-bit submodule doesn't accept whitespaces in the `Set` values, so `code` is
-		// used as a placeholder. This can be updated when the fix arrives to the current fluent-bit submodule
-		// `https://github.com/fluent/fluent-bit/issues/4286`.
```

Comment:
Is this comment not relevant still?

Reply:
For this specific case it is not relevant anymore, since the change https://github.com/fluent/fluent-bit/pull/6387 allows to parse spaces correctly in `values` (for example `Set a "b c"` now works correctly) when set within quotes `"..."`. This change was brought into the Ops Agent in https://github.com/GoogleCloudPlatform/ops-agent/pull/1559 .

I updated the referred fluent-bit issue https://github.com/fluent/fluent-bit/issues/4286 too.